src/data/value/mod.rs

```diff
@@ -283,6 +284,25 @@ impl Value {
             _ => Err(ValueError::UnaryMinusOnNonNumeric.into()),
         }
     }
+
+    pub fn like(&self, other: &Value) -> Result<Value> {
+        use Value::*;
+
+        match (self, other) {
+            (Str(a), Str(b)) => Ok(Bool(Regex::new(
+                &format!(
+                    "^{}$",
+                    regex::escape(b)
+                        .replace("%", ".*")
+                        .replace("_", ".")
+                )
+            )
+            .map_err(|e| ValueError::FailedToParsePattern(e.to_string()))?
+            .is_match(a)
+        )),
+            _ => Ok(Bool(false)),
```

Comment:
`_ => Ok(Bool(false)),` branch needs to be tested.
```sql
SELECT name FROM Boss WHERE name LIKE 10;
```
will work

Reply:
I realize that for other data types currently are can give mislead behaviour, when any other non text use of `LIKE` operator are return a `false` result.
Maybe instead of gives false result, it is more appropriate gives error such like `unsupported operator with data type ...`
Again, this as observation in postgres behaviour on case such like this, which gives errors :
```
ERROR: operator does not exist: text ~~ integer
```

Reply:
That really makes sense, I missed it. I'm also for postgres approach.
Also for the consistency, because current `Value` and `Literal` returns error for unsupported binary or unary operations, it would be also better to return error than returning `false` result.
Thanks for mentioning this!

Ok, then how about adding something like `ValueError::LikeOnNonString(Value, Value)` and `LiteralError::LikeOnNonString(String, String)`, and also adding `tests/filter.rs` have these two error test cases?